### PR TITLE
Multi build

### PR DESCRIPTION
**Pull request by zxan1285**
## ✏️ Changes
Now we can build logging extensions (logs-to-*) separately. Use `npm run extension:build -- --provider=mixpanel` to build single extension, or `npm run extension:build` to build all the extensions. `npm run build` will build UI and all the extensions. 

screen shot 2018-11-19 at 15 14 25
All extensions, each with own `webtask.json` and one common UI.

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  process.exit(0);
+}());
+
+function buildProvider (providerName) {
+  console.log(`Building logs-to-${providerName} extension...`);
+
+  prepareDirectory(providerName);
+  npmRun.sync(`rimraf dist/${providerName}`);
+  const command = `npm run prepareWebtask -- --provider=${providerName}&&` +
```
Comment (fyockm) [inline build.js]: missing space? `--provider=${providerName} &&`

File: .gitignore [hunk 2]
```diff
@@ -36,3 +36,5 @@ node_modules
 .idea
 server/data.json
 server/config.json
+dist
+webtask.json
```
Comment (fyockm) [inline .gitignore]: why added here?

File: server/routes/html.js [hunk 3]
```diff
@@ -53,7 +53,10 @@ module.exports = () => {
     };
 
     // Render from CDN.
-    const clientVersion = process.env.CLIENT_VERSION;
+    const { A0EXT_PROVIDER, COMMON_CLIENT_VERSION } = process.env;
+    const clientVersion = COMMON_CLIENT_VERSION
+      ? COMMON_CLIENT_VERSION[A0EXT_PROVIDER] || COMMON_CLIENT_VERSION.default
+      : null;
```
Comment (fyockm) [inline server/routes/html.js]: what happens if `CLIENT_VERSION` is null?

File: webtask-templates/common.json [hunk 4]
```diff
@@ -49,7 +44,7 @@
     "LOG_LEVEL": {
       "description": "This allows you to specify the log level of events that need to be sent",
       "type": "select",
-      "allowMultiple": true,
+      "allowMultiple": false,
```
[the remark this answers is not in the document]
Comment (zxan1285) [reply webtask-templates/common.json]: https://auth0team.atlassian.net/browse/KEY-352

File: server/lib/processLogs.js [hunk 5]
```diff
@@ -1,9 +1,9 @@
 const moment = require('moment');
 const loggingTools = require('auth0-log-extension-tools');
 
-const senders = require('./senders');
 const logger = require('./logger');
 const config = require('./config');
+const sendLogs = require(`./senders/${process.env.A0EXT_PROVIDER}`)();
```
[the remark this answers is not in the document]
Comment (zxan1285) [reply server/lib/processLogs.js]: If we're building separate extension for each provider, we don't want every extension bundle to contain code of all other extensions. That's why we're "requiring" only one `sender` at one time. This is working fine with `auth0-extensions-cli@1.0.11`. And for running locally you need to add `A0EXT_PROVIDER` environment variable before running `serve:dev`

File: .gitignore (hunk 2 above)
Comment (zxan1285) [reply .gitignore]: There are separate `webtask.json` files for each of the extensions. In `./dist/provider-name` dir. `./webtask.json` will be rewritten on each extension build.

File: server/routes/html.js (hunk 3 above)
Comment (zxan1285) [reply server/routes/html.js]: That means extension is running locally.

File: package.json [hunk 6]
```diff
@@ -70,14 +70,20 @@
       "auth0-log-extension-tools",
       "express-conditional-middleware",
       "pino"
-    ]
+    ],
+    "settings": {
+      "COMMON_CLIENT_VERSION": {
```
[the remark this answers is not in the document]
Comment (zxan1285) [reply package.json]: All logging extensions are using one client. By my idea, we should place common logging-extension-ui assets to the cdn, and use it in each of our logging-extensions. Each extension should use latest version of common client `COMMON_CLIENT_VERSION.default`. But we might need exception. For example, we're updating only `mixpanel` to use new version of client, and the rest of extension isn't ready yet. We can simply set `COMMON_CLIENT_VERSION.mixpanel = "2.0.0"`.

[the remark this answers is not in the document]
Comment (zxan1285) [reply package.json]: On second thought, we could just copy UI files to `./dist/provider-name` if we want each extension to have its own client (with same version as extension itself).

[the remark this answers is not in the document]
Comment (zxan1285) [reply package.json]: We decided to remove `COMMON CLIENT VERSION` and use direct link in `server/routes/html.js`, as we won't (supposedly) be changing UI often enough to use this setting.

**Pull request merged**